contracts/EthStorageContractL2Mode1.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "./EthStorageContract1.sol";
+
+/// @title IL1Block
+/// @notice Interface for L1Block contract.
+interface IL1Block {
+    /// @notice Get the blockhash of an L1 history block number.
+    /// @param _historyNumber The L1 history block number.
+    /// @return The blockhash of the L1 history block number.
+    function blockHash(uint256 _historyNumber) external view returns (bytes32);
+
+    /// @notice Get the current L1 block number.
+    /// @return The current L1 block number.
+    function number() external view returns (uint64);
+
+    /// @notice Get the current L1 block timestamp.
+    /// @return The current L1 block timestamp.
+    function timestamp() external view returns (uint64);
+}
+
+/// @title ISoulGasToken
+/// @notice Interface for the SoulGasToken contract.
+interface ISoulGasToken {
+    function chargeFromOrigin(uint256 _amount) external returns (uint256);
+}
+
+/// @custom:proxied
+/// @title EthStorageContractL2Mode1
+/// @notice EthStorage contract that will be deployed on L2, and uses L1Block contract to mine.
+contract EthStorageContractL2Mode1 is EthStorageContract1 {
```

Comment:
Is it possible for both L2 contracts to inherit from a common base contract to share most of the code?